opm/simulators/linalg/bda/WellContributions.cpp

```diff
@@ -56,12 +81,30 @@ WellContributions::~WellContributions()
         delete ms;
     }
     multisegments.clear();
-}
 
+#if HAVE_CUDA
+    if(cuda_gpu){
+        freeCudaMemory();
```

Comment:
Now you have reintroduced the bug that I recently fixed in #2720 . Was there a special reason to change the order? This needs to move back to line 46!

Reply:
So the `freeCudaMemory()` function should come before the delete statements which follow `if (h_x) {` ? Is that it?

Reply:
Qutoing from the PR

>With Multisegment wells we allocate WellContributions::hx and hy with CudaMallocHost. Yet we tried to deallocate them with delete[]. This caused segementation faults e.g. for model1/MSW_MODEL_1. Now we use CudaFreeHost for freeing if we used CUDA.

To have `if (h_x)` be false the pointer needs to be freed with `freeCudaMemory` before. Would be nice if you would put  a comment there (which I should probably habve done).

Reply:
Got it. Fixed it.